pkg/karmadactl/register/register.go

```diff
@@ -509,7 +509,7 @@ func (o *CommandRegisterOption) constructKarmadaAgentConfig(bootstrapClient *kub
 	}
 
 	klog.V(1).Infof("Waiting for the client certificate to be issued")
-	err = wait.Poll(1*time.Second, o.Timeout, func() (done bool, err error) {
+	err = wait.PollUntilContextTimeout(context.TODO(), 1*time.Second, o.Timeout, false, func(ctx context.Context) (done bool, err error) {
```

Comment:
Why `immediate` here is set to `false`? `condition` can be invoked immediately.

Reply:
That is because this behavior is equipment with previous.
`wait.Poll` always waits the interval before the run of `condition`.

Reply:
That's the idea not run the condition so rush.